### PR TITLE
[SDP-202] ensure non-logged-in users can access projects list, details, and search.

### PR DESCRIPTION
**Pull request by johndavid400**

### CODE REVIEW

File: app/views/projects/show.html.haml [hunk 1]
```diff
@@ -74,10 +75,11 @@
...
 
-= link_to 'Edit', edit_project_path(@project) if current_user.can_manage_project?(@project)
-\|
+= link_to 'Edit', edit_project_path(@project) if ProjectPolicy.new(current_user, @project).edit?
+
 = link_to 'Back', projects_path
 
-.add_attendee
-  = render 'add_attendee', project: @project
+- if ProjectPolicy.new(current_user, @project).edit?
```
Comment (cjarrett) [inline app/views/projects/show.html.haml]: any user should be able to add themselves to a project

File: app/policies/admin/project_policy.rb [hunk 2]
```diff
@@ -46,6 +46,10 @@ def destroy?
     index?
   end
 
+  def can_manage?
+    @user && @user.can_manage_project?(@project)
```
Comment (cjarrett) [inline app/policies/admin/project_policy.rb]: I ran into an unchecked nil for @project when I was testing. Maybe add a check for the presence of a project in can_manage_project?

File: app/views/projects/show.html.haml (hunk 1 above)
Comment (johndavid400) [reply app/views/projects/show.html.haml]: sure enough.... this should be .add_attendee not .edit

File: app/policies/admin/project_policy.rb (hunk 2 above)
Comment (johndavid400) [reply app/policies/admin/project_policy.rb]: fixed this in the user model method can_manage_project?

Comment (johndavid400) [reply app/policies/admin/project_policy.rb]: https://github.com/highlands/serveapp/pull/111/commits/5ded3278dac61d0d58274ec83a458cf3c86591d9

**Pull request merged**